gen/opennebula

```diff
@@ -0,0 +1,131 @@
+#!/usr/bin/perl
+
+use strict;
+use warnings;
+use perunServicesInit;
+use perunServicesUtils;
+use Perun::Agent;
+use JSON::XS;
+
+our $SERVICE_NAME = "opennebula";
+our $PROTOCOL_VERSION = "3.0.0";
+my $SCRIPT_VERSION = "3.0.0";
+
+#forward declaration
+sub processGroupData; 
+sub processMembersData; 
+sub processResourceData;
+
+perunServicesInit::init;
+my $DIRECTORY = perunServicesInit::getDirectory;
+my $data = perunServicesInit::getHierarchicalData;
+
+#Constants
+our $A_USER_MAIL;                  *A_USER_MAIL =                 \'urn:perun:user:attribute-def:def:preferredMail';
+our $A_USER_DISPLAY_NAME;          *A_USER_DISPLAY_NAME =         \'urn:perun:user:attribute-def:core:displayName';
+our $A_USER_ON_SSH_KEYS;           *A_USER_ON_SSH_KEYS =          \'urn:perun:user:attribute-def:virt:openNebulaSSHAdminKeys';
+our $A_U_CERT_DNS;                 *A_U_CERT_DNS  =               \'urn:perun:user:attribute-def:virt:userCertDNs';
+our $A_U_KERBEROS_LOGINS;          *A_U_KERBEROS_LOGINS =         \'urn:perun:user:attribute-def:virt:kerberosLogins';
+our $A_M_STATUS;                   *A_M_STATUS =                  \'urn:perun:member:attribute-def:core:status';
+our $A_USER_LOGIN;                 *A_USER_LOGIN =                \'urn:perun:user_facility:attribute-def:virt:login';
+our $A_U_F_BLACKLISTED;            *A_U_F_BLACKLISTED =           \'urn:perun:user_facility:attribute-def:virt:blacklisted';
+our $A_RESOURCE_ON_GROUP_NAME;     *A_RESOURCE_ON_GROUP_NAME =    \'urn:perun:resource:attribute-def:def:openNebulaGroupName';
+our $A_RESOURCE_ON_PRIVILEGED;     *A_RESOURCE_ON_PRIVILEGED =    \'urn:perun:resource:attribute-def:def:openNebulaPrivilegedUsers';
+
+my $dataByUser = {};
+
+foreach my $resourceData ($data->getChildElements) {
+	processResourceData $resourceData;
+}
+
+my @users = values %$dataByUser;
+my %outStruc = ( "users" => \@users, );
+
+my $fileName = "$DIRECTORY/$SERVICE_NAME";
+open FILE,">$fileName" or die "Cannot open $fileName: $! \n";
+print FILE JSON::XS->new->utf8->pretty->encode(\%outStruc);
```

Comment:
You can get json ouput sorted by keys through whole structure using ``...->pretty->canonical->encode(...``. It helps with future debug of changes in generated data.